src/components/ShoppingList/dataPreparation.ts

```diff
@@ -0,0 +1,222 @@
+import {
+    fetchClients,
+    fetchFamilies,
+    fetchLists,
+    fetchParcels,
+} from "@/components/ShoppingList/databaseFetch";
+import { Schema } from "@/supabase";
+import { Person } from "@/components/Form/formFunctions";
+
+export interface ShoppingListProps {
+    postcode: string;
+    parcelInfo: ParcelInfo;
+    clientSummary: ClientSummary;
+    householdSummary: HouseholdSummary;
+    requirementSummary: RequirementSummary;
+    itemsList: Item[];
+    endNotes: string;
+}
+
+export interface ParcelInfo {
+    Voucher_Number: string;
+    Packing_Date: string;
+    Collection_Date: string;
+    Collection_Site: string;
+}
+
+export interface ClientSummary {
+    Name: string;
+    Contact: string;
+    Address: string;
+    Extra_Information: string;
+}
+
+export interface HouseholdSummary {
+    Household_Size: string;
+    Gender_Breakdown: string;
+    Number_Of_Babies: string;
+    Age_and_Gender_of_Children: string;
+}
+
+export interface RequirementSummary {
+    Feminine_Products_Required: string;
+    Baby_Products_Required: string;
+    Pet_Food_Required: string;
+    Dietary_Requirements: string;
+    Other_Items: string;
+}
+
+export interface Item {
+    description: string;
+    quantity: string;
+    notes: string;
+}
+
+const formatDate = (dateString: string | null): string => {
+    if (dateString === null) {
+        return "";
+    }
+    return new Date(dateString).toLocaleString("en-GB", {
+        year: "numeric",
+        month: "numeric",
+        day: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+    });
+};
+
+const prepareParcelInfo = async (
+    parcelID: string
+): Promise<{ parcelInfo: ParcelInfo; clientID: string }> => {
```

Comment:
It's probably worth interfacing this

Reply:
And all the rest of these in the file. (personally I reckon it's probably fine when only used once but I've been told otherwise in other reviews)